### DISCUSSION TITLE
Rewrite code on file save

### DISCUSSION THREAD

**Discussion by AlexeyDuybo** [ast-grep/ast-grep#829] [Q&A]
Hi, is it possible to rewrite the code, according with the "fix" option in the rules, when saving the file? I use vscode extension but it doesnt rewrite the files on save

**Comment by HerringtonDarkholme** [comment]
Hi! It is not possible at the moment since the format on save needs support from editor plugin.

But it is a nice to have feature! Let me open an issue on ast-grep-vscode to track this request. Thanks for the suggestion!

**Reply by HerringtonDarkholme** [reply]
Tracked in https://github.com/ast-grep/ast-grep-vscode/issues/67.

I also think it is worth having a rule-level option to control if the auto fix should be done automatically. 
Not all fixes are safe to commit automatically.
